### PR TITLE
Use OpenSSL 3.1.5 and 3.2.1

### PR DESCRIPTION
**Pull request by GrahamCampbell**

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
Oh, lol. Related: https://github.com/quictls/openssl/issues/156.

File: .github/workflows/linux.yml [hunk 1]
```diff
@@ -47,7 +47,7 @@ env:
   mod_h2-version: v2.0.26
   msh3-version: v0.6.0
   openssl3-version: openssl-3.1.3
```
Comment (jay) [inline .github/workflows/linux.yml:49]: should this one be updated as well?

**Comment by jay** [comment]
blocked by openssl/openssl#23693

~~~
test 0313...[CRL test]
 valgrind ERROR ==95535== 1,795 (248 direct, 1,547 indirect) bytes in 1 blocks are definitely lost in loss record 32 of 32
==95535==    at 0x4848899: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==95535==    by 0x4F4A440: CRYPTO_malloc (in /home/runner/openssl3/lib/libcrypto.so.3)
==95535==    by 0x4F4A59D: CRYPTO_zalloc (in /home/runner/openssl3/lib/libcrypto.so.3)
==95535==    by 0x4E04218: asn1_item_embed_new (in /home/runner/openssl3/lib/libcrypto.so.3)
==95535==    by 0x4E01344: asn1_item_embed_d2i (in /home/runner/openssl3/lib/libcrypto.so.3)
==95535==    by 0x4E020E2: ASN1_item_d2i_ex (in /home/runner/openssl3/lib/libcrypto.so.3)
==95535==    by 0x50240F8: PEM_ASN1_read_bio (in /home/runner/openssl3/lib/libcrypto.so.3)
==95535==    by 0x509337D: X509_load_crl_file (in /home/runner/openssl3/lib/libcrypto.so.3)
==95535==    by 0x492EC06: populate_x509_store (openssl.c:3326)
==95535==    by 0x492F153: Curl_ssl_setup_x509_store (openssl.c:3479)
==95535==    by 0x492AA71: ossl_bio_cf_in_read (openssl.c:774)
==95535==    by 0x4E107FA: bread_conv (in /home/runner/openssl3/lib/libcrypto.so.3)
==95535==    by 0x4E0E5B6: BIO_read (in /home/runner/openssl3/lib/libcrypto.so.3)
==95535==    by 0x4CB1CB9: tls_default_read_n (in /home/runner/openssl3/lib/libssl.so.3)
==95535==    by 0x4CB203E: tls_get_more_records (in /home/runner/openssl3/lib/libssl.so.3)
==95535==    by 0x4CB0EE9: tls_read_record (in /home/runner/openssl3/lib/libssl.so.3)
==95535== 
~~~

**Comment by pheiduck** [comment]
I the meantime OpenSSL 3.3.0 is out.

**Pull request closed without merging**